app/controllers/intakes_controller.rb

```diff
@@ -10,13 +10,14 @@ def new
   def create
     @intake = Intake.new(intake_params.except(:voucher).merge(
       have_you_ever_experienced_homelessness_before:
-        ActiveRecord::Type::Boolean.new.cast(intake_params[:homelessness_first_time]),
+        !ActiveRecord::Type::Boolean.new.cast(intake_params[:homelessness_first_time]),
       non_cash_benefits: intake_params[:non_cash_benefits].reject {|r| r == "0" }
     ))
 
     client_params = intake_params[:client_attributes]
     @client = Client.new(client_params.merge(
-      race: client_params[:race].reject {|r| r == "0" }
+      race: client_params[:race].reject {|r| r == "0" },
+      veteran_separation_year: client_params[:veteran_separation_year].presence
```

Comment:
bit of a workaround for rails' form builder text_field sending empty string instead of nil when no value